src/profile/helpers/editData.ts

```diff
@@ -315,6 +315,23 @@ function updateItemAndCloneList(
   return newList;
 }
 
+function setAllItemsToRemoveAndCloneList(
+  allItems: EditData[],
+  item: EditData
+): EditData[] {
+  const index = findItemIndex(allItems, item.id);
+  if (index < 0) {
+    throw new Error('Item not found in updateAllItemsAndCloneList() ');
+  }
+
+  const newList = _.cloneDeep(allItems).map(element => ({
```

Comment:
Noticed there is a Sonar suggestion that we can return this immediately:
https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&pullRequest=349&id=City-of-Helsinki_open-city-profile-ui&open=AY-BUD0YbXIfC8RMhKz7